ext/GradedArraysTensorAlgebraExt/GradedArraysTensorAlgebraExt.jl

```diff
@@ -1,109 +1,78 @@
 
-using BlockArrays: Block, BlockIndexRange, blockedrange, blocks
-using BlockSparseArrays:
-  BlockSparseArrays,
-  AbstractBlockSparseArray,
-  AbstractBlockSparseArrayInterface,
-  BlockSparseArray,
-  BlockSparseArrayInterface,
-  BlockSparseMatrix,
-  BlockSparseVector,
-  block_merge
-using DerivableInterfaces: @interface
+using BlockArrays: blocks
+using BlockSparseArrays: BlockSparseArray, blockreshape
+using GradedArrays: GradedArray
 using GradedArrays.GradedUnitRanges:
-  GradedUnitRanges,
   AbstractGradedUnitRange,
   blockmergesortperm,
   blocksortperm,
-  dual,
+  flip,
   invblockperm,
-  nondual,
   unmerged_tensor_product
-using LinearAlgebra: Adjoint, Transpose
+using GradedArrays.SymmetrySectors: trivial
 using TensorAlgebra:
-  TensorAlgebra, FusionStyle, BlockReshapeFusion, SectorFusion, fusedims, splitdims
-using TensorProducts: OneToOne
+  TensorAlgebra,
+  AbstractBlockPermutation,
+  BlockedTuple,
+  FusionStyle,
+  trivial_axis,
+  unmatricize
 
-#=
-    reducewhile(f, op, collection, state)
+struct SectorFusion <: FusionStyle end
 
-reducewhile(x -> length(x) < 3, vcat, ["a", "b", "c", "d"], 2; init=String[]) ==
-  (["b", "c"], 4)
-=#
-function reducewhile(f, op, collection, state; init)
-  prev_result = init
-  prev_state = state
-  result = prev_result
-  while f(result)
-    prev_result = result
-    prev_state = state
-    value_and_state = iterate(collection, state)
-    isnothing(value_and_state) && break
-    value, state = value_and_state
-    result = op(result, value)
-  end
-  return prev_result, prev_state
-end
-
-#=
-    groupreducewhile(f, op, collection, ngroups)
+TensorAlgebra.FusionStyle(::Type{<:GradedArray}) = SectorFusion()
 
-groupreducewhile((i, x) -> length(x) ≤ i, vcat, ["a", "b", "c", "d", "e", "f"], 3; init=String[]) ==
-  (["a"], ["b", "c"], ["d", "e", "f"])
-=#
-function groupreducewhile(f, op, collection, ngroups; init)
-  state = firstindex(collection)
-  return ntuple(ngroups) do group_number
-    result, state = reducewhile(x -> f(group_number, x), op, collection, state; init)
-    return result
-  end
-end
+# TODO consider heterogeneous sectors?
+TensorAlgebra.trivial_axis(t::Tuple{Vararg{AbstractGradedUnitRange}}) = trivial(first(t))
 
-TensorAlgebra.FusionStyle(::AbstractGradedUnitRange) = SectorFusion()
+fuse_or_trivial_axis(axes::Tuple, ::Tuple) = unmerged_tensor_product(axes...)
+fuse_or_trivial_axis(::Tuple{}, axes::Tuple) = trivial_axis(axes)
 
-# Sort the blocks by sector and then merge the common sectors.
-function block_mergesort(a::AbstractArray)
-  I = blockmergesortperm.(axes(a))
-  return a[I...]
+function row_and_column_axes(
+  blocked_axes::BlockedTuple{2,<:Any,<:Tuple{Vararg{AbstractUnitRange}}}
+)
+  codomain_axes, domain_axes = blocks(blocked_axes)
+  @assert !(isempty(codomain_axes) && isempty(domain_axes))
+  row_axis = fuse_or_trivial_axis(codomain_axes, domain_axes)
+  unflipped_col_axis = fuse_or_trivial_axis(domain_axes, codomain_axes)
```

Comment:
Does this work?
```suggestion
  row_axis = unmerged_tensor_product(trivial_axes(domain_axes), codomain_axes...)
  unflipped_col_axis = unmerged_tensor_product(trivial_axes(codomain_axes), domain_axes...)
```
Just a way to avoid defining `fuse_or_trivial_axis`, that function was a bit hard for me to understand.